public/app/features/datasources/components/DataSourcesListHeader.tsx

```diff
@@ -19,7 +20,15 @@ const sortOptions = [
 
 export function DataSourcesListHeader() {
   const dispatch = useDispatch();
-  const setSearchQuery = useCallback((q: string) => dispatch(setDataSourcesSearchQuery(q)), [dispatch]);
+  const setSearchQuery = useCallback(
+    (q: string) => {
+      dispatch(setDataSourcesSearchQuery(q));
+      if (q) {
+        trackDsSearched({ query: q });
```

Comment:
we should debounce this as it will send a request on every key stroke - it should be ideally -> send a request when done typing.